ml_regression.ipynb

```diff
@@ -0,0 +1,347 @@
+{
```

Comment:
As these are more function definitions (or class as you are creating), I would have created a ml_regression.py file, and define them there. I think we envision Jupyter notebooks to be used as sample codes/visualization of how your functions work :)